fabric-enchanting-api-v1/src/main/java/net/fabricmc/fabric/mixin/enchanting/MixinEnchantmentHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.enchanting;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.enchantment.EnchantmentTarget;
+import net.minecraft.enchantment.InfoEnchantment;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.enchanting.v1.FabricEnchantment;
+
+@Mixin(EnchantmentHelper.class)
+public class MixinEnchantmentHelper {
+	@Unique
+	private static Enchantment currentEnchantment;
```

Comment:
thread safety notes. Hope no one calls this from non-server thread and break things

Reply:
> thread safety notes. Hope no one calls this from non-server thread and break things

Do you have any alternative solutions? Sadly mixin doesn't allow to capture locals in redirects.

At least chances would be quite low as `getHighestApplicableEnchantmentsAtPower` would have to be called simultaneously.